apis/r/R/TileDBObject.R

```diff
@@ -15,7 +15,8 @@ TileDBObject <- R6::R6Class(
     #' @param internal_use_only Character value to signal this is a 'permitted' call,
     #' as `new()` is considered internal and should not be called directly.
     initialize = function(uri, platform_config = NULL, tiledbsoma_ctx = NULL,
-                          tiledb_timestamp = NULL, internal_use_only = NULL) {
+                          tiledb_timestamp = NULL, internal_use_only = NULL,
+                          soma_context = NULL) {
```

Comment:
I thought we had agreed on not exposing `soma_context()` via `$new()` at this point in time https://github.com/single-cell-data/TileDB-SOMA/pull/2988#discussion_r1757629865

Reply:
Yes.  This gives the object to also set one here but it is not a requirement.  Having the entry point is good for testing, methinks.  See the comment a few lines futher down as well as the need to rework / let go of `tiledbsoma_ctx`.  I think this is all best addressed in a next PR.

https://github.com/single-cell-data/TileDB-SOMA/blob/091e1094380998766995eb1f2bac9ebdff33b35b/apis/r/R/TileDBObject.R#L34-L51